src/main/scala/shine/OpenCL/SeparateHostAndKernelCode.scala

```diff
@@ -0,0 +1,127 @@
+package shine.OpenCL
+
+import arithexpr.arithmetic.NamedVar
+import shine.DPIA._
+import shine.DPIA.Phrases._
+import shine.DPIA.Types._
+import shine.OpenCL.primitives.functional.{KernelCall, Run}
+import shine.OpenCL.primitives.imperative.OpenCLKernelDefinition
+import scala.collection.mutable
+
+import scala.language.existentials
+import scala.annotation.tailrec
+
+object SeparateHostAndKernelCode {
+  type SizedKernelDef = (LocalSize, GlobalSize, OpenCLKernelDefinition)
```

Comment:
As this is part of the interface (and therefore not a local type) I think it would make sense to put this elsewhere and make it a case class (or to add an Option for Local/Global size to `OpenCLKernelDefinition`.

Reply:
I added the size information to the kernel definition.